get_url pass incorrect If-Modified-Since header(#67417)

wangzw opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Fix #67417. HTTP header value of `If-Modified-Since` set by `get_url` does not follow HTTP protocol.
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
get_url

sivel reviewed (approved):
Please also provide a changelog fragment in `changelogs/fragments`

wangzw commented:
> Please also provide a changelog fragment in `changelogs/fragments`

Added

wangzw committed:
get_url pass incorrect If-Modified-Since header(#67417)

Fix #67417. HTTP header value of `If-Modified-Since` set by `get_url` does not follow HTTP protocol.

jamescassell reviewed (approved):
shipit

Though I wonder whether UTC would be better.

relrod commented:
rebuild_merge

relrod commented:
/rebuild
